lib/OpenQA/Worker/Jobs.pm

```diff
@@ -164,6 +175,21 @@ sub upload {
 
     my $res = $OpenQA::Worker::Common::ua->start($tx);
 
+    # Upload known server failures (Instead of anything that's not 200)
+    unless ($res->res->code =~ m/^5/) {
```

Comment:
Agree with @foursixnine, `unless` (even in the block form) is idiomatic Perl. Even though in this case he could have used `unless ($res->res->is_status_class(500)) {`, and maybe renamed that `$res` to something that looks better. :wink: